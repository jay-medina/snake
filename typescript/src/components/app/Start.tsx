import React from 'react';
import { PlayButton } from './PlayButton';

export interface StartProps {
  onPlayClick: () => void;
}

export const Start: React.SFC<StartProps> = ({ onPlayClick }) => {
  return (
    <div className="snake__start-screen">
      <div className="snake__start-screen-title">Snake</div>
      <PlayButton onClick={onPlayClick} />
    </div>
  );
};
