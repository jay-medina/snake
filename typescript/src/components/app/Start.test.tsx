import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Start, StartProps } from './Start';

describe('Start', () => {
  let component: ShallowWrapper<StartProps>;
  let onPlayClick: jest.Mock;

  beforeEach(() => {
    onPlayClick = jest.fn();
    component = shallow(<Start onPlayClick={onPlayClick} />);
  });

  it('displays the component', () => {
    expect(component).toMatchSnapshot();
  });

  describe('when user clicks on the play button', () => {
    beforeEach(() => {
      component.find('PlayButton').simulate('click');
    });

    it('invokes the onPlayClick callback', () => {
      expect(onPlayClick).toHaveBeenCalled();
    });
  });
});
