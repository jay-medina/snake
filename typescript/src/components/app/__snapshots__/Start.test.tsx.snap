// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Start displays the component 1`] = `
<div
  className="snake__start-screen"
>
  <div
    className="snake__start-screen-title"
  >
    Snake
  </div>
  <PlayButton
    onClick={[MockFunction]}
  />
</div>
`;
